### DISCUSSION TITLE
Improving Documentation

### DISCUSSION THREAD

**Discussion by collinsonindo** [pgagroal/pgagroal#368] [Ideas]
Hi, Collins here.

I did apply for gsoc 23 for user vault and was perusing through docs and code trying to at least understand a bit more of how the codes work and felt that there is an overwhelming amount of "assumed knowledge" needed to understand the inner workings, which a first timer may lack .

I propose to spend some time at least documenting what is existing practice here (and other bouncers) to at least understand how such systems work, regardless whether my proposal is accepted or not.

I suggest the docs to be placed in the pgagroal website if it meets some criteria for greater availability to the masses who may need to understand such systems like me.

cc @jesperpedersen  just for a heads up.

**Comment by jesperpedersen** [comment]
Any additional documentation - on web site or in code - would be great.

Web site is in `gh-pages`

**Comment by collinsonindo** [comment]
Cool, lemme start working on it.

Okay if i start with overviews of how such things work?

**Reply by jesperpedersen** [reply]
Yeah, of course. Anything will help people out.

**Comment by ianhxu** [comment]
Hi, this is a great idea! I also applied for gsoc 23 but for query cache. I'm adding code comments while getting familiar with the code. I can also format and PR my comments for others to use.

May I open an issue for adding comments and documentation in the code? @jesperpedersen

**Reply by jesperpedersen** [reply]
Once you have a pull request ready in your local repository then create the associated issue
